### ISSUE TITLE
Failed to resolve "MvvmCross.Core.MvxSetup" reference from "MvvmCross, Version=6.1.0.0, Culture=neutral, PublicKeyToken=null"

### ISSUE THREAD

**Issue by diegostamigni** [MvvmCross/MvvmCross#2956]
TL;DR I am currently experiencing the following issue:
```
MTOUCH: Error MT2002: Failed to resolve "MvvmCross.Core.MvxSetup" reference from "MvvmCross, Version=6.1.0.0, Culture=neutral, PublicKeyToken=null" (MT2002) (Drive.iOS)
```

Long story short: we are using MvvmCross since the beginning of the project, and recently we added to that project an iOS extension (Share extension), that, of course, is using MvvmCross. Suddenly tho, after the upgrade from Xamarin.iOS 11.9 to 11.12 it now started failing with the error above and the only solution to fix this was to downgrade back to 11.9.

An interesting thing to note here is that if I remove the Share extension as a reference from the main iOS project (that is, I exclude it from the build process and final binary) it compiles fine.

Any idea why?

Setup:
- iOS main project
- PCL libraries
- iOS extension project

Xamarin.iOS 11.12, Mono 5.10, macOS 10.13.5

### Steps to reproduce :scroll:
Unfortunately, I am not able to reproduce this, tried in a new project and it worked fine, probably _is not complex enough_?

### Expected behavior :thinking:
Should build successfully

### Actual behavior :bug:
```
MTOUCH : warning MT0109: The assembly 'System.Net.Http.dll' was loaded from a different path than the provided path (provided path: /Users/osiris/.nuget/packages/system.net.http/4.3.3/runtimes/win/lib/netstandard1.3/System.Net.Http.dll, actual path: /Library/Frameworks/Xamarin.iOS.framework/Versions/11.12.0.4/lib/mono/Xamarin.iOS/System.Net.Http.dll).
MTOUCH : warning MT0127: Incremental builds have been disabled because this version of Xamarin.iOS does not support incremental builds in projects that include more than one third-party binding libraries.
MTOUCH : warning MT0113: Native code sharing has been disabled for the extension 'Drive.iOS.ShareProvider' because the --assembly-build-target options are different between the container app () and the extension (--assembly-build-target:@all=dynamiclibrary).
MTOUCH : error MT2002: Failed to resolve "MvvmCross.Core.MvxSetup" reference from "MvvmCross, Version=6.1.0.0, Culture=neutral, PublicKeyToken=null"
    63 Warning(s)
    1 Error(s)
```

### Configuration :wrench:

**Version:** 6.x

**Platform:** 
- [x] :iphone: iOS
- [ ] :robot: Android
- [ ] :checkered_flag: WPF
- [ ] :earth_americas: UWP
- [ ] :apple: MacOS
- [ ] :tv: tvOS
- [ ] :monkey: Xamarin.Forms

**Comment by Cheesebaron** [comment]
What does the reference in your extension project look like? Which dll is it using?

**Comment by diegostamigni** [comment]
@Cheesebaron What do you mean ? MvvmCross is same across the whole solution, v6.1.0 installed via nuget.

**Comment by Cheesebaron** [comment]
Look in the csproj and figure out whether the path is correct to the referenced assembly. If say you are referencing the wrong one, for a different platform, then you might have issues. Maybe also check that the assembly gets copied to the output folder too.

**Comment by diegostamigni** [comment]
Ah, I see what you mean here @Cheesebaron: a whole lot of dlls (including MvvmCross) is missing from the output folder of the iOS extension... argh
